roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -150,4 +167,50 @@ class RooMinimizer : public TObject {
   ClassDefOverride(RooMinimizer,0) // RooFit interface to ROOT::Fit::Fitter
 } ;
 
+// include here to avoid circular dependency issues in class definitions
+#include "TestStatistics/MinuitFcnGrad.h"
```

Comment:
As mentioned in the comment, it's a circular dependency. `RooMinimizer`'s template constructor needs a complete `MinuitFcnGrad` type and `MinuitFcnGrad`'s template constructor needs a complete `RooMinimizer` type.

Other options to solve this would be:
1. Split the headers into two: one file with only the class definition and one file with template and inline functions. Not so nice, imho.
2. Put all classes into one header (really bad idea).
3. Tear out the `RooMinimizer` dependency in `MinuitFcnGrad`'s constructor. Strictly speaking, it doesn't need the `RooMinimizer`, it uses it to get to the `Fitter` (which contains the `FitConfig` (which contains the parameter settings object that it *actually* needs)). I could just pass the thing it actually needs to the constructor. The slight downside of this is that it is important that the correct set of parameters is passed, and getting them directly from the `RooMinimizer`, as done now, ensures this. If the ctor caller has to pass this correct set in manually, there's a risk this will be done wrong some day.

I went with option 3 now, since actually the ctor is not the only point where the correct set of parameters has to be passed in, Synchronize has the same "issue".